src/envelope.rs

```diff
@@ -114,7 +182,17 @@ impl<A: Handler<M>, M: Send + 'static> MessageEnvelope for NonReturningEnvelope<
         act: &'a mut Self::Actor,
         ctx: &'a mut Context<Self::Actor>,
     ) -> BoxFuture<'a, ()> {
-        Box::pin(act.handle(self.message, ctx).map(|_| ()))
+        let Self {
+            message,
+            instrumentation,
+            ..
+        } = *self;
+
+        Box::pin(
+            instrumentation
+                .apply::<A, M, _>(act.handle(message, ctx))
+                .map(|_| ()),
```

Comment:
Nit: We could remove this if we would restrict `Handler` to a `Return` of `()`.

Reply:
True, but as of this change it is still used with any `Return`, so this can wait for a followup I think